contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1673,6 +1675,200 @@ object_name(PG_FUNCTION_ARGS)
 	PG_RETURN_NULL();
 }
 
+/*
+ * type_id
+ * Returns the object ID with type name as input.
+ * Returns NULL
+ *  if input is NULL
+ *  if there is no such type
+ *  if user don't have right permission
+ *  if any error occured
+ */
+Datum
+type_id(PG_FUNCTION_ARGS)
+{
+    char       *db_name,
+               *schema_name,
+               *object_name;
+    char       *physical_schema_name;
+    char       *input;
+    char      **splited_object_name;
+    Oid         schema_oid;
+    Oid         user_id = GetUserId();
+    Oid result = InvalidOid;
+    int         i;
+
+    if (PG_ARGISNULL(0))
+        PG_RETURN_NULL();
+    input = text_to_cstring(PG_GETARG_TEXT_P(0));
+
+    /* strip trailing whitespace from input */
+    i = strlen(input);
+    while (i > 0 && isspace((unsigned char) input[i - 1]))
+        input[--i] = '\0';
+
+    /* length should be restricted to 4000 */
+    if (i > 4000)
+        ereport(ERROR,
+                (errcode(ERRCODE_STRING_DATA_LENGTH_MISMATCH),
+                 errmsg("input value is too long for object name")));
+
+    /* resolve the two part name */
+    splited_object_name = split_object_name(input);
+    /* If three part name(db_name also included in input) then return null */
+    if(strcmp(splited_object_name[1], ""))
+    {
+        pfree(input);
+        pfree(splited_object_name);
```

Comment:
splited_object_name is double pointer, clear out inner pointers first and then outer one